### PR TITLE
NF: datalad-osf

### PR DESCRIPTION
**Pull request by yarikoptic**
- [x] add secrets needed for testing into secrets of this repo

### CODE REVIEW

**Comment by yarikoptic** [comment]
will probably just need to wait for https://github.com/datalad/datalad-osf/pull/122 to be merged and released and then nothing custom would be needed here (although probably worthwhile keeping `pip_depends` functionality just in case, but remove its use from datalad-osf spec since it is not needed).

**Comment by yarikoptic** [comment]
@mih @adswa @bpoldrack - any interest in finishing up this PR or should it be just closed?

**Comment by adswa** [comment]
it would be nice to have it, I can try to have a look after Monday.

**Comment by adswa** [comment]
ah, I don't have permissions to add secrets to this repo. But I'll dig them out and send them to you @yarikoptic

**Comment by yarikoptic** [comment]
I have made you Admin on this repo @adswa

**Comment by adswa** [comment]
cool, added the secrets here. I think I also need to put them into env variables in .travis.yml. But will resolve conflicts first

**Comment by adswa** [comment]
Okay done @yarikoptic. Not that I have simply limited testing to datalad-osf by explicitly specifying datalad-osf.tests in the nose call. With that, https://github.com/datalad/datalad-osf/pull/122 becomes obsolete (I think that that PR also didn't manage to exclude the osf tests, the logs there are still full of "module osfclient not found" errors)

File: extensions.yaml [hunk 1]
```diff
@@ -31,6 +31,9 @@ extensions:
   pypi: datalad-webapp
   package: datalad_webapp
   tester: pytest
+ datalad_osf:
+  pip_depends: [pytest]
```
Comment (yarikoptic) [inline extensions.yaml:35]: why `pytest` dependency but then 
```
python -m nose -s -v --with-cov --cover-package datalad datalad_osf.tests
```
should `tester: pytest` be added if you use pytest?

Also note that the workflow is a "generated" file -- it must not be edited manually.

**Review by yarikoptic — COMMENTED** [review]
NB cannot request changes since I guess I am the one who started this PR

 the workflow file must not be edited manually, it should be generated out of spec.  I do not quite grasp why it is needed to specify `datalad_osf.tests`, so may be that https://github.com/datalad/datalad-osf/pull/122 is not really obsolete? ;)  if any module is missing you of cause could add it to `pip_depends` to be installed, but it should be the `datalad-osf` which would declare such dependency (either for main installation in install_requires, or for testing in `devel` extra)

File: extensions.yaml (hunk 1 above)
Comment (adswa) [reply extensions.yaml:35]: > Also note that the workflow is a "generated" file -- it must not be edited manually.

How are secrets supposed to be added? They need to go into the workflow file as far as I know?
https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets

Comment (yarikoptic) [reply extensions.yaml:35]: GOOD POINT!  
- Ideally we should allow in the spec to list secrets to expose and provision for that in the template. So everything alike for any other optionally included part of the workflow, e.g. for apt_depends: https://github.com/datalad/datalad-extensions/blob/master/templates/.github/workflows/test-%7B%7Bextension.name%7D%7D.yaml#L69
- meanwhile, if not feeling up to implementing above (but I really hope that you don't succumb to this ;)), I guess there should be no harm if we expose secrets even within unrelated workflows by just tuning the main template to always include them https://github.com/datalad/datalad-extensions/blob/master/templates/.github/workflows/test-%7B%7Bextension.name%7D%7D.yaml#L76

**Comment by mih** [comment]
Maybe close this one and merge #80, which is green?

**Pull request closed without merging**